### DISCUSSION TITLE
Spell Check and Word Suggestions

### DISCUSSION THREAD

**Discussion by terry9873** [florisboard/florisboard#1924] [Q&A]
Hi, I don't understand how to use Github very well. But I have spent time reading a lot of threads. Some say spell checker and word suggestions should be working in 0.3.14 (I have 0.3.16 via Droidify), some say it shouldn't, some say it will for some but not for others. I have read that rolling back to an older version can make it work. Patrick Gold made a comment on one issue thread that spell check only works with English US...

Re that last point, I am using English UK language settings in my OS, if I change to English US will that enable spell checking?

I have only just started using Floris Board after I asked a friend to recommend a privacy respecting KB with top number row. I was BLOWN AWAY by this KB, cut and paste, cursor movement, WOW!! It's GORGEOUS, how did I ever live without it?! But then I noticed no spell checking. I do not need suggestions really, I know what I want to type, but spell checker I badly need and can't do without as I can rarely hit the correct keys (I am old and have fat fingers!) Is there any a way to get this working by downgrading (not sure how) as suggested above, or by a switch of language or any other trick? if not, any ideas roughly when it will be working? I will have to switch back to default OS KB until I can get spell checker on Floris. 

PS THANK YOU for such a great app, I will be contributing/supporting with more than just words once I am using it full time :)

**Comment by terry9873** [comment]
Should spell check be working in Floris 0.3.16 ?

**Reply by tsiflimagas** [reply]
Yes, but it's not what you probably think. Spell checking is only a service that compares the typed word against a dictionary and underlines it red if it's not found there. It won't autocorrect the word, nor will it suggest any words. These features are separate, and they're not implemented yet.

**Reply by terry9873** [reply]
ah i see thanks very much. Where can I keep an eye out for updates for when autocorrect/suggest is implemented? Is there a thread I can watch for that? thanks again

**Reply by tsiflimagas** [reply]
[Here](https://github.com/florisboard/florisboard/issues/1921) there are some information about it, and perhaps more will be written as the process progress. It will be good if you keep an eye also on the releases page, as alpha builds will be released every now and then for users to try the progress of the work on word suggestions, and release notes will be having information about what's new.

**Reply by terry9873** [reply]
thanks, but where/what is the releases page?

**Reply by tsiflimagas** [reply]
Oh, you're indeed new to GitHub XD. No problem, you'll get used to it, it's not that hard.
Here's the releases page https://github.com/florisboard/florisboard/releases. Here new builds/versions of FlorisBoard are released. To not confuse you, you can get them either from there or from the other available places, F-Droid, IzzyOnDroid and Playstore, if you enrol in the beta program.

**Reply by terry9873** [reply]
thank you. i have Droidify, will find the beta program and try that. thanks again

**Comment by terry9873** [comment]
I'd be really grateful for any kind of response here.

**Reply by yarub123** [reply]
@tsiflimagas thanks for responding to Terry. Im on the newest alpha but im not seeing any spell checking with a red underline? I have the default samsung keyboard as the spell checker. Also when text field is empty or i have not typed in any letters, the smartbar fills up with the word "next",  like so: 
![Screenshot_20230111_033352_Firefox](https://user-images.githubusercontent.com/84487964/211796143-a6e8e411-a6f8-4345-80fe-19cedc6ba08d.jpg)

Also I too am a noob. (Yes i know so far the autosuggestions feature isnt implemented and we will be shot if we ask about it 😅)
